crates/nu-parser/src/parser.rs

```diff
@@ -1432,6 +1444,7 @@ pub fn parse_range(
     //   and  <range_operator> is ".." or "..<"
     //   and one of the <from> or <to> bounds must be present (just '..' is not allowed since it
     //     looks like parent directory)
+    //bugbug range cannot be [..] because that looks like parent directory
```

Comment:
Do we properly address this through parse ordering or through defining `SyntaxShape::Filepath` for `cd` etc. and doing type/signature directed parsing?

Reply:
The bugbug is still relevant -- e.g, Nu complains about `| range ..`, but accepts `| range 1..`.
I am not the parser expert you're looking for, but I do think a fix will require more than simply re-doing the order of tests for shapes.  Range help says its signature is:
```
Signatures:
  list<any> | range <any> -> list<any>

```
Argument <any> cannot be the right shape.

Reply:
OK, thanks for adding the comment to keep us honest.